### PR TITLE
feat(worker): added opentelemetry 

### LINKED ISSUE #975: Add opentelemetry to our API
Our API runs on cloudflare workers, the entry point is [/apps/api/src/worker.ts](/https://github.com/unkeyed/unkey/blob/main/apps/api/src/worker.ts)

We need better tracing insights and want to use [otel-cf-workers](https://github.com/evanderkoogh/otel-cf-workers) to trace:
1. The request
2. Durable Objects
3. Cloudflare queues
4. Caches
5. Database queries via drizzle + planetscale.js [file](https://github.com/unkeyed/unkey/blob/main/apps/api/src/pkg/db.ts)

As sink for the traces, we can use either https://baselime.io or https://axiom.co

___

Only default spans are in scope for this issue, you don't need to add any spans manually, though they must work of course.

### PR DESCRIPTION
**Pull request by mehulmathur16**
/claim #975
/fixes #975

## What does this PR do?

Integrated opentelemetry.

fixes #975 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

This PR is completely tested. You just need to insert your baselime api key and service name in .dev.vars file. After doing this make some api calls like verifying keys. And finally after some time [15-20 mins], the traces will be visible on baselime console.

`pnpm fmt` is showing some warnings in the files that I have not made any changes to.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [ ] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

### CODE REVIEW

**Comment by ubinatus** [comment]
Have you tried running a request against the API? Using the auto-instrumentation breaks the server. It lets you spin up but fails when interacting with the db:

```
api:dev: ✘ [ERROR] Unhandled error while verifying key {"error":"Cannot read properties of undefined (reading 'fetch')","keyHash":"p4/dfWk9dC+34zYsepEiuSsANcemDeL1dxto9sUfJSA="}
api:dev: 

api:dev: ✘ [ERROR] TypeError: Cannot read properties of undefined (reading 'fetch')
api:dev: 
api:dev:       at null.<anonymous> (file:///Users/ubinatus/Personal/unkey/node_modules/.pnpm/@microlabs+otel-cf-workers@1.0.0-rc.20/node_modules/@microlabs/otel-cf-workers/src/instrumentation/fetch.ts:241:80)
```

Somehow, `fetch` is not defined when it reaches the drizzle connector. This is an actual issue with the package itself.

[getActiveConfig() returns null #76](https://github.com/evanderkoogh/otel-cf-workers/issues/76)

**Comment by mehulmathur16** [comment]
> Have you tried running a request against the API? Using the auto-instrumentation breaks the server. It lets you spin up but fails when interacting with the db:
> 
> ```
> api:dev: ✘ [ERROR] Unhandled error while verifying key {"error":"Cannot read properties of undefined (reading 'fetch')","keyHash":"p4/dfWk9dC+34zYsepEiuSsANcemDeL1dxto9sUfJSA="}
> api:dev: 
> 
> api:dev: ✘ [ERROR] TypeError: Cannot read properties of undefined (reading 'fetch')
> api:dev: 
> api:dev:       at null.<anonymous> (file:///Users/ubinatus/Personal/unkey/node_modules/.pnpm/@microlabs+otel-cf-workers@1.0.0-rc.20/node_modules/@microlabs/otel-cf-workers/src/instrumentation/fetch.ts:241:80)
> ```
> 
> Somehow, `fetch` is not defined when it reaches the drizzle connector. This is an actual issue with the package itself.
> 
> [getActiveConfig() returns null #76](https://github.com/evanderkoogh/otel-cf-workers/issues/76)

Hi @ubinatus, this isn't tested yet.

**Comment by chronark** [comment]
@mehulmathur16 how did you deploy this?
I'm getting:
```bash
 pnpm wrangler --env=preview deploy
 ⛅️ wrangler 3.28.3
-------------------

✘ [ERROR] Expected string in JSON but found "type"

    ../../node_modules/.pnpm/github.com+mehulmathur16+otel-cf-workers@e11cfed425f800552835e34096740d0444e5401c/node_modules/@microlabs/otel-cf-workers/dist/esm/package.json:1:1:
      1 │ {type: module}
        ╵  ~~~~


✘ [ERROR] Expected string in JSON but found "type"

    node_modules/@microlabs/otel-cf-workers/dist/esm/package.json:1:1:
      1 │ {type: module}
        ╵  ~~~~


✘ [ERROR] Build failed with 2 errors:

  ../../node_modules/.pnpm/github.com+mehulmathur16+otel-cf-workers@e11cfed425f800552835e34096740d0444e5401c/node_modules/@microlabs/otel-cf-workers/dist/esm/package.json:1:1:
  ERROR: Expected string in JSON but found "type"
  node_modules/@microlabs/otel-cf-workers/dist/esm/package.json:1:1: ERROR: Expected string in JSON
  but found "type"

```

Looks like your custom fork is not working correctly?

**Comment by mehulmathur16** [comment]
@chronark I had fixed this issue yesterday here : https://github.com/evanderkoogh/otel-cf-workers/compare/main...mehulmathur16:otel-cf-workers:main

Can you please check you had the latest pull when reinstalling modules ?

File: apps/api/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@unkey/keys": "workspace:^",
     "@chronark/zod-bird": "0.3.3",
     "@hono/zod-openapi": "^0.9.6",
+    "@microlabs/otel-cf-workers": "github:mehulmathur16/otel-cf-workers",
```
Comment (ubinatus) [inline apps/api/package.json:27]: IMO this should be hosted as an npm package, especially for security reasons, just if the Unkey team decides to go with this solution.

**Pull request closed without merging**